app/notifications/process_notifications.py

```diff
@@ -182,16 +182,56 @@ def transform_notification(
     return notification
 
 
-def db_save_notification(notification: Notification):
+def db_save_and_send_notification(notification: Notification):
     dao_create_notification(notification)
     if notification.key_type != KEY_TYPE_TEST:
         if redis_store.get(redis.daily_limit_cache_key(notification.service_id)):
             redis_store.incr(redis.daily_limit_cache_key(notification.service_id))
+    current_app.logger.info(f"{notification.notification_type} {notification.id} created at {notification.created_at}")
+
+    deliver_task = choose_deliver_task(notification)
+    try:
+        deliver_task.apply_async([str(notification.id)], queue=notification.queue_name)
+    except Exception:
+        dao_delete_notifications_by_id(notification.id)
+        raise
```

Comment:
Hmm I didn't remember that part of the code where we delete a notification, because we could not send it to the queue? 🤔